### PR TITLE
Remove `NotImplementedError` in resample with rule 

### LINKED ISSUE #6271: Resample fails on different rules
**What happened**:
`resample` raises `NotImplementedError` error for rules that don't evenly divide a day.  

**What you expected to happen**:
Different frequency rules should be allowed for time series resampling. 

**Minimal Complete Verifiable Example**:

```python
p = 100
index = pd.date_range(start="2012-01-02", periods=p, freq="H")
df = pd.DataFrame({"p": np.random.random(p)}, index=index)
ddf = dd.from_pandas(df, npartitions=5)
result = ddf.resample("2D").count().compute()
--------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
~/workplaces/dask-env/dask/dask/dataframe/tseries/resample.py in __init__(self, obj, rule, **kwargs)
    132                 "Resampling frequency %s that does"
    133                 " not evenly divide a day is not "
--> 134                 "implemented" % rule
    135             )
    136         self._rule = rule

NotImplementedError: Resampling frequency <2 * Days> that does not evenly divide a day is not implemented
```

**Anything else we need to know?**:
I got a correct resampling for the example above after removing the check on:
https://github.com/dask/dask/blob/79836fe4a52c9cf0e1eb32525792ad6bfed1c62b/dask/dataframe/tseries/resample.py#L130

Also for test:
https://github.com/dask/dask/blob/79836fe4a52c9cf0e1eb32525792ad6bfed1c62b/dask/dataframe/tseries/tests/test_resample.py#L131

```python
expected = s.resample('57T').mean()
expected
```
```
2012-01-02 00:00:00    28
2012-01-02 00:57:00    78
Freq: 57T, dtype: int64
```

```python
ds = dd.from_pandas(s, npartitions=5)
result = ds.resample('57T').mean().compute()
result
```
```
2012-01-02 00:00:00    28
2012-01-02 00:57:00    78
Freq: 57T, dtype: int64
```
I can attempt a PR to remove raising this error, if my assumption here is correct. 

**Environment**:

- Dask version: 2.17.2
- Python version: 3.7.6
- Operating System: MacOS 10.15
- Install method (conda, pip, source): source

### PR DESCRIPTION
**Pull request by abduhbm**
This PR closes #6271

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/dataframe/tseries/tests/test_resample.py [hunk 1]
```diff
@@ -128,12 +128,29 @@ def test_resample_pads_last_division_to_avoid_off_by_one():
     assert_eq(actual, expected)
 
 
-def test_series_resample_not_implemented():
+def test_resample_does_not_evenly_divide_day():
+    import numpy as np
+
+    p = 100
+    index = pd.date_range(start="2012-01-02", periods=p, freq="H")
```
Comment (jsignell) [inline dask/dataframe/tseries/tests/test_resample.py:135]: In other places in this test file we create indexes with holes in them:

https://github.com/dask/dask/blob/38d85be356ba7bfc0bf561881525b75b7a5a7e7e/dask/dataframe/tseries/tests/test_resample.py#L29-L30

Can you do something like that in these?

Comment (abduhbm) [reply dask/dataframe/tseries/tests/test_resample.py]: Added. Thanks for the review @jsignell

**Comment by jcrist** [comment]
Test failure is unrelated. LGTM, merging. Thanks @abduhbm!

**Comment by jsignell** [comment]
Congrats @abduhbm on your first PR in dask :)

**Pull request merged**